be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -1212,6 +1213,12 @@ Status SegmentIterator::_lookup_ordinal_from_pk_index(const RowCursor& key, bool
 
     // The sequence column needs to be removed from primary key index when comparing key
     bool has_seq_col = _segment->_tablet_schema->has_sequence_col();
+    bool has_rowid = !_segment->_tablet_schema->cluster_key_idxes().empty();
```

Comment:
Is it possiable to add a new page type for IndexedColumnWrter that can write key:value into it, using the rowid as value.
In this case, we don't need to process logics of rowid here.